pkg/actuators/machine/machine_scope.go

```diff
@@ -73,7 +75,10 @@ func newMachineScope(params machineScopeParams) (*machineScope, error) {
 		providerStatus:     providerStatus,
 	}
 
-	mscp.getNutanixCredentials()
+	if err := mscp.getNutanixCredentials(); err != nil {
+		return nil, fmt.Errorf("failed to get the credentials to access the Nutanix PC: %v", err.Error())
```

Comment:
Nit: It's more idiomatic in go just to wrap the error, the callers will handle this correctly so the message when printed is correct, but it also means if the error is typed, this can be detected in a parent scope
```suggestion
	if err := mscp.getNutanixCredentials(); err != nil {
		return nil, fmt.Errorf("failed to get the credentials to access the Nutanix PC: %w", err)
```